### ISSUE TITLE
Not registering controllerless blocks and can't override Invoke method on BlockComponent

### ISSUE THREAD

**Issue by jacob-g** [episerver/netcore-preview#34]
I set up an empty project (using `dotnet new epicmsempty --name ProjectName`) and am no able to create controllerless blocks. If I attempt to do so, they do not load at all.

If I try to create a block with a controller, it only works if I leave the class empty (i.e. do not provide an actual controller method), since it will use the inherited `PartialContentComponent::Invoke`. However, if I try to create an actual controller method, I get this error:
![image](https://user-images.githubusercontent.com/1960707/115919387-62f3bb00-a43e-11eb-96d5-59dd1b59f45f.png)

For reference, here is the relevant code:
```csharp
public class BravoBlockComponent : BlockComponent<BravoBlock>
{
    private IBravoConfig _bravoConfig;

    public BravoBlockComponent(IBravoConfig bravoConfig)
    {
        _bravoConfig = bravoConfig ?? throw new ArgumentNullException(nameof(bravoConfig));
    }
}

[ContentType(DisplayName = "HTML Block", GUID = "{35A51CFE-AA4A-455F-8EB7-DF5ED4328E39}")]
public class HtmlBlock : SiteBlockData
{
    [Display(Name = "Title", GroupName = SiteGroupNames.Content, Order = EpiConstants.ContentStartIndex + EpiConstants.ItemDelta * 0)]
    public virtual string Title { get; set; }

    [Display(Name = "Content", GroupName = SiteGroupNames.Content, Order = EpiConstants.ContentStartIndex + EpiConstants.ItemDelta * 1)]
    public virtual XhtmlString Content { get; set; }
}

public class SiteBlockData : BlockData
{
}
```

The view (in Views/Shared/Components/HtmlBlock/Default.cshtml)
```razor
@model HtmlBlock

<div>
    @if (Model.Title is not null)
    {
        <h3>@Model.Title</h3>
    }
    @Html.PropertyFor(m => m.Content)
</div>
```

**Comment by lunchin** [comment]
Thanks, added to the internal tracker

**Comment by lunchin** [comment]
This has been fixed in will be in cms 12.0.3
